### PR TITLE
ENH: Update dipy, vtk, ubuntu version

### PR DESCRIPTION
**Pull request by AntoineTheb**
Update VTK to 9.0.1 as previous version was too old
Update to dipy 1.2
Remove python3.5 since it is no longer supported by dipy
Update ubuntu version to 18 (bionic)

@arnaudbore @frheault @mdesco @GuillaumeTh Ready for review

### CODE REVIEW

**Comment by frheault** [comment]
@karanphil The test will fail due to ssst modif.
Can you sync with Antoine to provide the proper fix AND then delete your PR when this one is merged?

**Comment by AntoineTheb** [comment]
@karanphil The files have been added to the PR. Can you test to make sure that the combination of dipy 1.2 and your improvements work ? Or can you provide data so I can test as well ?

**Comment by karanphil** [comment]
> @karanphil The files have been added to the PR. Can you test to make sure that the combination of dipy 1.2 and your improvements work ? Or can you provide data so I can test as well ?

@AntoineTheb , you copied the wrong files! Did you copied from this https://github.com/karanphil/scilpy/tree/adapting_ssst_csd?

**Comment by AntoineTheb** [comment]
Argh that's embarassing lol, I indeed copied the wrong files. Will fix asap

**Comment by AntoineTheb** [comment]
To be completely sure, these are the files to be modified ?
https://github.com/scilus/scilpy/compare/master...karanphil:adapting_ssst_csd

**Comment by karanphil** [comment]
> To be completely sure, these are the files to be modified ?
> [master...karanphil:adapting_ssst_csd](https://github.com/scilus/scilpy/compare/master...karanphil:adapting_ssst_csd)

Yes!

**Comment by AntoineTheb** [comment]
Should be the right files now @karanphil

**Comment by karanphil** [comment]
My changes seem to work just fine!

**Comment by frheault** [comment]
Well this PR is a roller coaster!

File: requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ bz2file==0.98.*
 coloredlogs==10.0.*
 cycler==0.10.*
 Cython==0.29.*
-dipy==1.1.*
+dipy>=1.1.2
```
Comment (frheault) [inline requirements.txt:5]: 1.2.*

File: scripts/scil_compute_ssst_frf.py [hunk 2]
```diff
@@ -65,10 +65,12 @@ def _build_arg_parser():
              'fiber voxels in the automatic estimation. [%(default)s]')
 
     p.add_argument(
-        '--roi_radius', default=10, type=int,
-        help='If supplied, use this radius to select single fibers from the '
-             'tensor to estimate the FRF. The roi will be a cube spanning '
-             'from the middle of the volume in each direction. [%(default)s]')
+        '--roi_radii', default=[10], nargs='+', type=int,
+        help='If supplied, use those radii to select a cuboid roi '
```
Comment (frheault) [inline scripts/scil_compute_ssst_frf.py:69]: Maybe this is a bit verbose, maybe keep the description about a cube only? Or move some of it to the description? @karanphil can you help design something a bit shorter for Antoine?

File: scripts/scil_compute_ssst_frf.py [hunk 3]
```diff
@@ -92,6 +94,13 @@ def main():
     assert_inputs_exist(parser, [args.input, args.bvals, args.bvecs])
     assert_outputs_exist(parser, args, args.frf_file)
 
+    if len(args.roi_radii) == 1:
+        roi_radii = args.roi_radii[0]
+    elif len(args.roi_radii) == 2:
+        parser.error('--roi_radii cannot be of size (2,).')
+    else:
```
Comment (frheault) [inline scripts/scil_compute_ssst_frf.py:101]: Do an if == 1, an if == 3 and else crash.

**Review by frheault — APPROVED** [review]
Other than my small comments, all good

**Review by frheault — APPROVED** [review]
If you address at least my if comments, that's all good

File: scripts/scil_plot_mean_std_per_point.py [hunk 4]
```diff
@@ -53,7 +53,7 @@ def main():
         for metric, metric_stats in bundle_stats.items():
             nb_points = len(metric_stats)
             num_digits_labels = len(str(nb_points))
-            means = []
+           means = []
```
Comment (frheault) [inline scripts/scil_plot_mean_std_per_point.py:56]: indentation

**Comment by AntoineTheb** [comment]
:tada:

File: scripts/scil_compute_ssst_frf.py [hunk 5]
```diff
@@ -92,6 +94,13 @@ def main():
     assert_inputs_exist(parser, [args.input, args.bvals, args.bvecs])
     assert_outputs_exist(parser, args, args.frf_file)
 
+    if len(args.roi_radii) == 1:
+        roi_radii = args.roi_radii[0]
+    elif len(args.roi_radii) == 3:
+        roi_radii = args.roi_radii
+    else:
+        parser.error('--roi_radii cannot be of size (2,).')
```
Comment (karanphil) [inline scripts/scil_compute_ssst_frf.py:102]: Maybe say 'wrong size for --roi_radii.'? As the size could be anything else than 1 or 3.

Comment (AntoineTheb) [reply scripts/scil_compute_ssst_frf.py]: Fixed, let me know what you think

Comment (karanphil) [reply scripts/scil_compute_ssst_frf.py]: Looks good to me!

**Pull request merged**